channels/rdpdr/client/rdpdr_main.c

```diff
@@ -1652,6 +1652,13 @@ static UINT rdpdr_virtual_channel_event_disconnected(rdpdrPlugin* rdpdr)
 {
 	UINT error;
 
+	/* Remove all dependent devices before tearing down resources */
+	if (rdpdr->devman)
```

Comment:
@bjcollins That is why I mentioned the ```ListDictionary_Clear(devman->devices); }``` function. It removes all devices and calls the cleanup function registered without eliminating the list (which is assumed existing)